nft/schema/rule.go

```diff
@@ -249,6 +316,45 @@ func (e *Expression) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+func (f Flags) MarshalJSON() ([]byte, error) {
+	var dynamicStruct interface{}
+
+	switch flagCount := len(f.Flags); {
+	case flagCount == 1:
+		dynamicStruct = f.Flags[0]
+	case flagCount > 1:
+		dynamicStruct = f.Flags
+	}
+
+	return json.Marshal(dynamicStruct)
+}
+
+func (f *Flags) UnmarshalJSON(data []byte) error {
+	var dynamicStruct interface{}
+	if err := json.Unmarshal(data, &dynamicStruct); err != nil {
+		return err
+	}
+
+	switch dynamicStruct.(type) {
+	case string:
+		d := dynamicStruct.(string)
+		f.Flags = []string{d}
+	case []interface{}:
+		slice := dynamicStruct.([]interface{})
+		for _, val := range slice {
+			stringVal, ok := val.(string)
+			if !ok {
+				return fmt.Errorf("flags values require string type: %T(%v)", dynamicStruct, dynamicStruct)
+			}
+			f.Flags = append(f.Flags, stringVal)
+		}
+	default:
+		return fmt.Errorf("flags values require string type: %T(%v)", dynamicStruct, dynamicStruct)
```

Comment:
You don't need to "cast" it again the switch statement can return the value already

```suggestion
	switch v := dynamicStruct.(type) {
	case string:
		f.Flags = []string{v}
	case []interface{}:
		for _, val := range v {
			stringVal, ok := val.(string)
			if !ok {
				return fmt.Errorf("flags values require string type: %T(%v)", dynamicStruct, dynamicStruct)
			}
			f.Flags = append(f.Flags, stringVal)
		}
	default:
		return fmt.Errorf("flags values require string type: %T(%v)", dynamicStruct, dynamicStruct)
```

Reply:
Nice, I did not know this language feature. Thanks!
Done